modules/serverless-logic-overview-timeouts.adoc

```diff
@@ -0,0 +1,205 @@
+
+If the Event state is a workflow starting state, incoming events might trigger workflow instances. In this case, if the `exclusive` property is set to `true`, the `eventTimeout` property is ignored.
+
+If the `exclusive` property is set to false, in this case, the defined `eventTimeout` represents the time between arrival of specified events.
+
+For example, consider the following example:
+
+.Example for the event timeout
+[source,json]
+----
+{
+"states": [
+{
+    "name": "ExampleEventState",
+    "type": "event",
+    "exclusive": false,
+    "timeouts": {
+      "eventTimeout": "PT2M"
+    }
+    "onEvents": [
+        {
+            "eventRefs": [
+                "ExampleEvent1",
+                "ExampleEvent2"
+            ],
+            "actions": [
+              ...
+            ]
+        }
+    ],
+    "end": {
+        "terminate": true
+    }
+}
+]
+}
+----
+
+The first `eventTimeout` starts after any of the referenced events are consumed. If the second event does not occur within the defined eventTimeout, no workflow instance should be created.
+
+Event timeouts supports callback state timeout, switch state timeout, and event state timeout.
+
+The `Callback` state can be used when you must execute an action in general to call an external service, and wait for an asynchronous response in the form of an event.
+
+.Example of Callback state with timeout
+[source,json]
+----
+{
+ "name": "CallbackState",
+ "type": "callback",
+ "action": {
+   "name": "callbackAction",
+   "functionRef": {
+     "refName": "callbackFunction",
+     "arguments": {
+       "input": "${\"callback-state-timeouts: \" + $WORKFLOW.instanceId + \" has executed the callbackFunction.\"}"
+     }
+   }
+ },
+ "eventRef": "callbackEvent",
+ "transition": "CheckEventArrival",
+ "onErrors": [
+   {
+     "errorRef": "callbackError",
+     "transition": "FinalizeWithError"
+   }
+ ],
+ "timeouts": {
+   "eventTimeout": "PT30S"
+ }
+}
+----
+
+You can use the `Switch` state when you need to take an action depending on certain conditions. These conditions can be based on the workflow data, `dataConditions`, or on events, `eventConditions`.
+
+.Example of Switch state with timeout
+[source,json]
+----
+{
+    "name": "ChooseOnEvent",
+    "type": "switch",
+    "eventConditions": [
+    {
+        "eventRef": "visaApprovedEvent",
+        "transition": "ApprovedVisa"
+    },
+    {
+        "eventRef": "visaDeniedEvent",
+        "transition": "DeniedVisa"
+    }
+    ],
+        "defaultCondition": {
+        "transition": "HandleNoVisaDecision"
+    },
+        "timeouts": {
+        "eventTimeout": "PT5S"
+    }
+}
+----
+
+The `Event` state is used to wait for one or more events to be received by the workflow, execute a set of actions, and then continue the execution. If the Event state is a starting state, a new workflow instance is created.
```

Comment:
```suggestion
The `Event` state is used to wait for one or more events to be received by the workflow, execute a set of actions, and then continue the execution. If the Event state is a starting state, a new workflow instance is created.

The timeouts property is used for this state to configure the maximum time the workflow should wait for the configured events to arrive.

If this time is exceeded and the events are not received, the workflow moves to the state defined in the transition property or ends the workflow instance (in case of an end state), without performing any actions.

```